### ISSUE TITLE
0.15.0dev0 CUDA CLI option parsing crash

### ISSUE THREAD

**Issue by xbkingx** [ethereum-mining/ethminer#923]
Getting generic crash (not responding) before any miner initialization when:
specifying --cuda-parallel-hash
specifying --cuda-block-size in addition to --cuda-grid-size, but NOT when specifying one or the other.

Tried each option individually and moving/separating options within the command, no difference.

Fails:
```
ethminer -RH -U -P addr1 -P addr2 -v 4 --display-interval 10 --cuda-schedule auto --cuda-grid-size 4096 --cuda-block-size 128 --cuda-parallel-hash 2 --cuda-streams 2
ethminer -RH -U -P addr1 -P addr2 -v 4 --display-interval 10 --cuda-schedule auto --cuda-parallel-hash 2 --cuda-streams 2
ethminer -RH -U -P addr1 -P addr2 -v 4 --display-interval 10 --cuda-schedule auto --cuda-grid-size 4096 --cuda-parallel-hash 2
ethminer -RH -U --cuda-grid-size 4096 -P addr1 -P addr2 -v 4 --display-interval 10 --cuda-schedule auto  --cuda-block-size 128
```

Works:
```
ethminer -RH -U -P addr1 -P addr2 -v 4 --display-interval 10 --cuda-schedule auto --cuda-grid-size 4096
ethminer -RH -U -P addr1 -P addr2 -v 4 --display-interval 10 --cuda-schedule auto --cuda-block-size 128
ethminer -RH -U -P addr1 -P addr2 -v 4 --display-interval 10 --cuda-schedule auto --cuda-parallel-hash 2
ethminer -RH -U -P addr1 -P addr2 -v 4 --display-interval 10 --cuda-schedule auto --cuda-streams 2 --cuda-grid-size 4096
ethminer -RH -U -P addr1 -P addr2 -v 4 --display-interval 10 --cuda-schedule auto --cuda-streams 2 --cuda-block-size 128
```

Info:
0.15.0.dev0+git.7c928e0 (also in previous artifact)
Win 10 x64 1709
GTX1080Ti (tested OC and stock)

**Comment by jean-m-cyr** [comment]
Tested on current master with no problem found:

ethminer -U -P stratum+ssl://${ETHWALLET}.$HOSTNAME@us1.ethermine.org:5555 -v 4 --display-interval 10 --cuda-schedule auto --cuda-grid-size 4096 
--cuda-block-size 128 --cuda-parallel-hash 2 --cuda-streams 2

ethminer -U -P stratum+ssl://${ETHWALLET}.$HOSTNAME@us1.ethermine.org:5555 -v 4 --display-interval 10 --cuda-schedule auto --cuda-parallel-hash 2
 --cuda-streams 2

ethminer -U -P stratum+ssl://${ETHWALLET}.$HOSTNAME@us1.ethermine.org:5555 -v 4 --display-interval 10 --cuda-schedule auto --cuda-parallel-hash 2

**Comment by xbkingx** [comment]
Huh... it DOES work - without the fallback pool address. You'd think the other options wouldn't matter. Try adding a fallback server. This still fails for me:

`ethminer.exe -RH -U -P stratum+tcp://<wallet>.<rig>@eth-us-east1.nanopool.org:9999 -P stratum+tcp://<wallet>.<rig>@eth-us-west1.nanopool.org:9999 -v 4 --display-interval 10 --cuda-schedule auto --cuda-grid-size 4096 --cuda-block-size 128 --cuda-parallel-hash 2 --cuda-streams 2`

but his works:

`ethminer.exe -RH -U -P stratum+tcp://<wallet>.<rig>@eth-us-east1.nanopool.org:9999 -v 4 --display-interval 10 --cuda-schedule auto --cuda-grid-size 4096 --cuda-block-size 128 --cuda-parallel-hash 2 --cuda-streams 2`

**Comment by jean-m-cyr** [comment]
Well it also works with a secondary on the latest master:

ethminer -U -P stratum+ssl://${ETHWALLET}.$HOSTNAME@us1.ethermine.org:5555 -P stratum+ssl://${ETHWALLET}.$HOSTNAME@us2.ethermine.org:5555 -v 4 --
display-interval 10 --cuda-schedule auto --cuda-block-size 128

I think this very recent commit was the fix: 1db84335b01ee194d363ad9bd337765aa5192b96
